src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -253,12 +289,28 @@ def set_up_managed_identity(self):
             self.session_pool_def["identity"] = identity_def
 
     def set_up_dynamic_configuration(self):
-        if self.get_argument_cooldown_period_in_seconds() is None:
+        if self.get_argument_lifecycle_type() is None:
+            # Auto infer lifecycle type
+            if self.get_argument_cooldown_period_in_seconds() is not None:
+                self.set_argument_lifecycle_type(LifecycleType.Timed.name)
+            elif self.get_argument_max_alive_period() is not None:
+                self.set_argument_lifecycle_type(LifecycleType.OnContainerExit.name)
+            else:
+                # Default to 'Timed'
+                self.set_argument_lifecycle_type(LifecycleType.Timed.name)
```

Comment:
We should set a default value for lifecycle-type in the description, then we don't need this:
```
    with self.argument_context('containerapp sessionpool create', arg_group='Configuration') as c:
        c.argument('lifecycle_type', arg_type=get_enum_type(["Timed", "OnContainerExit"]), help="The lifecycle type of the Session Pool", default='Timed')

    with self.argument_context('containerapp sessionpool update', arg_group='Configuration') as c:
        c.argument('lifecycle_type', arg_type=get_enum_type(["Timed", "OnContainerExit"]), help="The lifecycle type of the Session Pool")
```